feign-demo/src/test/java/com/nortal/test/feign/demo/glue/TestPageStepDefs.java

```diff
@@ -22,19 +22,98 @@
+
+    @When("user requests cat fact")
+    public void getCatFact() {
+        try {
+            final ResponseEntity<CatApi.Fact> responseEntity = catApi.getFact();
+            scenarioContext.putStepData(RESPONSE, responseEntity);
+        } catch (HttpClientErrorException ex) {
+            scenarioContext.putStepData(RESPONSE, ex);
+        }
+    }
+
+    @Then("user should get response with cat fact")
+    public void assertCatFact() {
+        ResponseEntity<CatApi.Fact> responseEntity = scenarioContext.getStepData(RESPONSE);
+        assertNotNull(responseEntity, "Response should not be null");
+        assertEquals(HttpStatus.OK, responseEntity.getStatusCode(), "Unexpected HTTP status code");
+
+        assertNotNull(responseEntity.getBody().getFact(), "Fact should not be null");
+        assertNotNull(responseEntity.getBody().getLength(), "Length should not be null");
+    }
+
+    @When("user requests a list of cat facts")
+    public void getCatFacts() {
+        try {
+            final ResponseEntity<CatApi.FactsList> responseEntity = catApi.getFacts();
+            scenarioContext.putStepData(RESPONSE, responseEntity);
+        } catch (HttpClientErrorException ex) {
+            scenarioContext.putStepData(RESPONSE, ex);
+        }
+    }
+
+    @Then("user should get response with a list of cat facts")
+    public void assertCatFacts() {
+        ResponseEntity<CatApi.FactsList> responseEntity = scenarioContext.getStepData(RESPONSE);
+        assertNotNull(responseEntity, "Response should not be null");
+        assertEquals(HttpStatus.OK, responseEntity.getStatusCode(), "Unexpected HTTP status code");
 
-    @Given("Api is called")
-    public void openPageInBrowser() {
-        ResponseEntity<CatApi.Fact> response = catApi.getFact();
+        assertEquals(CURRENT_PAGE, responseEntity.getBody().getCurrent_page(), "Unexpected current page");
+        assertNotNull(responseEntity.getBody().getLast_page(), "Last page should not be null");
+
+        assertTrue(responseEntity.getBody().getData().size() >= 1, "Unexpected number of facts");
+        responseEntity.getBody().getData().forEach(fact -> {
+            assertNotNull(fact.getFact(), "Fact should not be null");
+            assertNotNull(fact.getLength(), "Length should not be null");
+        });
     }
 
+    @When("user requests a list of cat breeds")
+    public void getBreeds() {
+        try {
+            final ResponseEntity<CatApi.BreedsList> responseEntity = catApi.getBreeds();
+            scenarioContext.putStepData(RESPONSE, responseEntity);
+        } catch (HttpClientErrorException ex) {
+            scenarioContext.putStepData(RESPONSE, ex);
+        }
+    }
 
+    @Then("user should get response with a list of cat breeds")
+    public void assertBreeds() {
+        ResponseEntity<CatApi.BreedsList> responseEntity = scenarioContext.getStepData(RESPONSE);
+        assertNotNull(responseEntity, "Response should not be null");
```

Comment:
Assertions are being duplicated, we could put them in a different method to clean up our code